galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/BackupCPS.java

```diff
@@ -0,0 +1,186 @@
+import org.apache.commons.logging.LogFactory;
+
+import org.osgi.service.component.annotations.Component;
+
+import dev.galasa.framework.FrameworkInitialisation;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+
+@Component(service = { BackupCPS.class })
+public class BackupCPS {
+    
+    private Log  logger  =  LogFactory.getLog(this.getClass());
+    
+    private final boolean         file_append       = false;
+    private final boolean         autoFlush         = true;
+    
+    String charset = "UTF-8";
+    
+    private File                    file;
+    private FileOutputStream        fos;
+    private OutputStreamWriter      osw;
+    private BufferedWriter          bw;
+    private PrintWriter             pw;
+    
+    /**
+     * <p>Retrieves CPS properties for all configured namespaces and sends them to standard output</p>
+     * 
+     * @param bootstrapProperties
+     * @param overrideProperties
+     * @param filePath
+     * @return
+     * @throws FrameworkException
+     */
+    public void backup(Properties bootstrapProperties, Properties overrideProperties, String filePath) throws FrameworkException {
+        
+        logger.info("Initialising CPS Backup Service");
+        
+        FrameworkInitialisation frameworkInitialisation = null;
+        try {
+            frameworkInitialisation = new FrameworkInitialisation(bootstrapProperties, overrideProperties);
+        } catch (Exception e) {
+            throw new FrameworkException("Unable to initialise the Framework Service", e);
+        }
+        
+        IFramework framework = frameworkInitialisation.getFramework();
+        IConfigurationPropertyStoreService cps = framework.getConfigurationPropertyService("framework");
+        
+        List<String> namespaces = cps.getCPSNamespaces(); 
+        
+        logger.info("Backing-up to file: " + filePath);
+        
+        try {
+            initialisePrintWriter(filePath);
+        } catch (Exception e) {
+            throw new FrameworkException("Error Initialising Print Writer", e);
+        }
+        
+        outputCPSProperties(namespaces, framework);
+        
+        closePrintWriter();
+        
+        logger.info("Ending CPS Backup Service");
+        
+    }
+    
+    /**
+     * <p>Takes a list of namespaces and outputs CPS properties for each namespace.</p>
+     * 
+     * @param namespaces
+     * @param framework
+     * @return
+     * @throws FrameworkException
+     */  
+    private void outputCPSProperties(List<String> namespaces, IFramework framework) throws FrameworkException {
+        logger.info("Backing Up Namespaces:");
+        for (String namespace : namespaces) {
+            if (isNamespaceBackupPermitted(namespace)) {
+                logger.info("SUCCESS:\t" + namespace);
+                outputNamespaceCPSProperties(namespace, framework);
```

Comment:
Question:    Does this output the properties in alphabetic order?  if not,  please do so.   Also,  I think it would be nice if after every namespace,  there is a blank line,

Reply:
Alphabetic output implemented, as well as blank lines between each namespace. Changes will be pushed shortly.